src/js/bootstrap-treeview.js

```diff
@@ -17,1233 +17,1206 @@
-				}, this));
-			}
-
-			// Continue selecting node
-			node.state.selected = true;
-			if (!options.silent) {
-				this.$element.trigger('nodeSelected', $.extend(true, {}, node));
-			}
-		}
-		else {
-
-			// Unselect node
-			node.state.selected = false;
-			if (!options.silent) {
-				this.$element.trigger('nodeUnselected', $.extend(true, {}, node));
-			}
-		}
-	};
-
-	Tree.prototype.toggleCheckedState = function (node, options) {
-		if (!node) return;
-		this.setCheckedState(node, !node.state.checked, options);
-	};
-
-	Tree.prototype.setCheckedState = function (node, state, options) {
-
-		if (state === node.state.checked) return;
-
-		if (state) {
-
-			// Check node
-			node.state.checked = true;
-
-			if (!options.silent) {
-				this.$element.trigger('nodeChecked', $.extend(true, {}, node));
-			}
-		}
-		else {
-
-			// Uncheck node
-			node.state.checked = false;
-			if (!options.silent) {
-				this.$element.trigger('nodeUnchecked', $.extend(true, {}, node));
-			}
-		}
-	};
-
-	Tree.prototype.setDisabledState = function (node, state, options) {
-
-		if (state === node.state.disabled) return;
-
-		if (state) {
-
-			// Disable node
-			node.state.disabled = true;
-
-			// Disable all other states
-			this.setExpandedState(node, false, options);
-			this.setSelectedState(node, false, options);
-			this.setCheckedState(node, false, options);
-
-			if (!options.silent) {
-				this.$element.trigger('nodeDisabled', $.extend(true, {}, node));
-			}
-		}
-		else {
-
-			// Enabled node
-			node.state.disabled = false;
-			if (!options.silent) {
-				this.$element.trigger('nodeEnabled', $.extend(true, {}, node));
-			}
-		}
-	};
-
-	Tree.prototype.render = function () {
-
-		if (!this.initialized) {
-
-			// Setup first time only components
-			this.$element.addClass(pluginName);
-			this.$wrapper = $(this.template.list);
-
-			this.injectStyle();
-
-			this.initialized = true;
-		}
-
-		this.$element.empty().append(this.$wrapper.empty());
-
-		// Build tree
-		this.buildTree(this.tree, 0);
-	};
-
-	// Starting from the root node, and recursing down the
-	// structure we build the tree one node at a time
-	Tree.prototype.buildTree = function (nodes, level) {
-
-		if (!nodes) return;
-		level += 1;
-
-		var _this = this;
-		$.each(nodes, function addNodes(id, node) {
-
-			var treeItem = $(_this.template.item)
-				.addClass('node-' + _this.elementId)
-				.addClass(node.state.checked ? 'node-checked' : '')
-				.addClass(node.state.disabled ? 'node-disabled': '')
-				.addClass(node.state.selected ? 'node-selected' : '')
-				.addClass(node.searchResult ? 'search-result' : '') 
-				.attr('data-nodeid', node.nodeId)
-				.attr('style', _this.buildStyleOverride(node));
-
-			// Add indent/spacer to mimic tree structure
-			for (var i = 0; i < (level - 1); i++) {
-				treeItem.append(_this.template.indent);
-			}
-
-			// Add expand, collapse or empty spacer icons
-			var classList = [];
-			if (node.nodes) {
```

Comment:
JavaScript returns 'true' even for empty arrays ([]).
Source: https://stackoverflow.com/questions/19146176/javascript-empty-array-evaluates-to-true-in-conditional-structures-why-is